### PR TITLE
Retry on 502, and do a channel reopen on UnknownHostException

### PR DESCRIPTION
**Pull request by droazen**

### CODE REVIEW

File: google-cloud-contrib/google-cloud-nio/src/main/java/com/google/cloud/storage/contrib/nio/CloudStorageRetryHandler.java [hunk 1]
```diff
@@ -147,7 +148,7 @@ static boolean isRetryable(final StorageException exs) {
     // Yes one wouldn't necessarily think of 403 as retriable,
     // but see https://github.com/broadinstitute/gatk/issues/3735
     // for reports of 403 being returned in cases when retrying works.
-    return exs.isRetryable() || exs.getCode() == 403 || exs.getCode() == 500 || exs.getCode() == 503;
+    return exs.isRetryable() || exs.getCode() == 403 || exs.getCode() == 500 || exs.getCode() == 503 || exs.getCode() == 502;
```
Comment (lbergelson) [inline google-cloud-contrib/google-cloud-nio/src/main/java/com/google/cloud/storage/contrib/nio/CloudStorageRetryHandler.java]: could you reformat these so they are in numerical order please

**Review by lbergelson — APPROVED** [review]
Dumb comment aside, this looks good.

File: google-cloud-contrib/google-cloud-nio/src/main/java/com/google/cloud/storage/contrib/nio/CloudStorageRetryHandler.java (hunk 1 above)
Comment (droazen) [reply google-cloud-contrib/google-cloud-nio/src/main/java/com/google/cloud/storage/contrib/nio/CloudStorageRetryHandler.java]: Done

**Review by lbergelson — APPROVED** [review]
LGTM

**Comment by droazen** [comment]
I also increased the maxDepth constant in `isReopenable()` from 10 to 20, since we were close to the limit in practice

**Pull request merged**